### ISSUE TITLE
Failed to find the StyleCop Assembly. Please check your StyleCop installation.

### ISSUE THREAD

**Issue by dziedrius** [kubiix/ReSharper.StyleCop#4]
I've installed StyleCop for Resharper 9 and it constantly is throwing:
"Failed to find the StyleCop Assembly. Please check your StyleCop installation."
I wen't to R# settings, StyleCop plugin settings, unmarked checkbox for "Automatically Check location" and set the path to stylecop's dll (btw, there was like 5 or 6 copies all over place) and it complained, that it should be "StyleCop.dll" not "StyleCop.ReSharper900.dll", so I renamed, but that didn't helped.
Also, I've tried to see where it searches for the dll using Process Monitor there was couple records with operation CreateFile and result "NAME NOT FOUND", but I'm not sure if that somehow relates to the problem.

**Comment by dziedrius** [comment]
I understood my problem. As long as previously R# integration came together with stylecop installation, there was no problem, now even the R# integration comes from R# extension manager, you need to install stylecop first. It would be nice though to get better error message, to understand why is it like that.

**Comment by itsmaryann** [comment]
I am getting the same error. I have uninstalled Resharper 9, deleted JetBrains folder from AppData\Local and AppData\Roaming, and then reinstalled, but still get the same issue. I did not understand how you fixed the issue. Did you first install StyleCop-4.7.49.0.msi (http://stylecop.codeplex.com/downloads/get/323236), then install Resharper 9, then install StyleCop from the Extensions?

**Comment by itsmaryann** [comment]
dziedrius, thanks for raising this issue and posting the solution! I was able to get rid of the error by installing StyleCop.

**Comment by baldwindc** [comment]
@itsmaryann Sounds like they got what they wanted :)

**Comment by SimonB251** [comment]
I still have this issue. I have tried installing in the R# extension manager, tried it with and without the StyleCop nuget to no avail, I have tried explicitly pointing to the StyleCop.dll in R# options and still no luck.

**Comment by wannado** [comment]
+1

**Comment by andysterks** [comment]
+1

**Comment by gligoran** [comment]
+1

**Comment by janhjordie** [comment]
+1

**Comment by toxato** [comment]
Has anyone fixed this issue? It is extremely annoying, getting the error message like 10 times each time i start Visual Studio.. ?

**Comment by TheMadKow** [comment]
Have you tried to delete the StyleCop plug-in on C:\Users\ {MyUser} \AppData\Local\JetBrains\plugins ?
It seems to stop the annoying errors on my end.

**Comment by Gennady-G** [comment]
Also met this error switching on stylecop in resharper extensions.
- googled "download stylecop"
- installed StyleCop-4.7.49.0 from first link
- restarted
- profit

**Comment by nwayve** [comment]
`choco install stylecop -y`

**Comment by citizenmatt** [comment]
I'd like to recommend using the "SyleCop by JetBrains" extension in the gallery. It solves several crashes, memory leaks and performance issues that are in this version, and it doesn't require the StyleCop Visual Studio extension to be installed. We're getting a lot of error reports from this version of the extension, and would like to move existing users to the "StyleCop by JetBrains" version (which is also open source at the [StyleCop/StyleCop](https://github.com/StyleCop/StyleCop) repo.

**Comment by Gennady-G** [comment]
Hi guys!

I described how I resolved this error here:
http://stackoverflow.com/questions/23639815/how-do-i-integrate-stylecop-4-7-with-visual-studio-2013/33561901#33561901

**Comment by nwayve** [comment]
Installing the "StyleCop by JetBrains" extension in the gallery was what I installed first.

My setup is a fresh install of Visual Studio 2015 CE (14.0.24720.0) w/ R# 10.0.2 Ultimate installed.  I installed xUnit.net Test Support 2.3.3 and ReSharper.StyleCop 4.7.60-alpha2, and restarted VS.  Upon restart, I received the barrage of alerts (Google brought me here).  I closed VS, did the `choco install stylecop -y`, which installed StyleCop 4.7.0, and resolved the alerts issue.

But this is not the preferred solution?

**Comment by citizenmatt** [comment]
No. The "ReSharper.StyleCop" extension is buggy. We're getting a lot of issues at JetBrains that are due to this extension. I'd strongly suggest uninstalling it, and installing "StyleCop by JetBrains". There are crashes, memory leaks and performance issues fixed. You can leave StyleCop 4.7.0 installed, or uninstall it. The "StyleCop by JetBrains" doesn't require it.

**Comment by nwayve** [comment]
:facepalm:
"ReSharper.StyleCop" != "StyleCop by JetBrains"
Got it. Uninstalled the first and installed the one by JetBrains.
